src/components/Input.tsx

```diff
@@ -1,18 +1,27 @@
 import { useState, useEffect } from "react";
-import { func, string } from "prop-types";
 import { getTrainInformation } from "../services/travel";
 import styles from "./Input.module.css";
+import { TrainResponse } from "../types/train";
 
-export default function Input({ placeholder, onSelect }) {
-  const [searchTerm, setSearchTerm] = useState("");
-  const [stationList, setStationList] = useState([]);
+type Props = {
+  placeholder: string;
+  onSelect: (trainCode: string) => void;
+};
+
+export default function Input({ placeholder, onSelect }: Props) {
```

Comment:
```suggestion
export default function StationAutocomplete({ placeholder, onSelect }: Props) {
```
If you want to have a specific input such as this one, I suggest you make the name of the component specific too. Unless, you want to make it reusable.
Btw, this component is autocomplete, it has more features than a simple input.